server/src/main/java/org/elasticsearch/index/mapper/LegacyGeoShapeFieldMapper.java

```diff
@@ -257,11 +257,10 @@ private void setupPrefixTrees(GeoShapeFieldType ft) {
         }
 
         private GeoShapeFieldType buildFieldType(BuilderContext context) {
-            GeoShapeFieldType ft = new GeoShapeFieldType(buildFullName(context), indexed, fieldType.stored(), false, meta);
+            GeoShapeFieldType ft =
+                new GeoShapeFieldType(buildFullName(context), indexed, fieldType.stored(), false, new LegacyGeoShapeParser(), meta);
```

Comment:
Can we share the same parser for the type and the mapper? I think we are doing that in other places.